utils/businesslogic.py

```diff
@@ -31,17 +30,42 @@ def new_transaction(transaction):
         Called when a new transaction has been added into database
         """
         logger.debug(f"New transaction {transaction}")
+
+        # Figure out a user for this transaction, if possible
         transaction_user = None
+
         if transaction.reference_number and transaction.reference_number > 0:
-            try:
-                # Figure out if a user can be mapped to transaction
-                transaction_user = CustomUser.objects.get(
+            # Search subscriptions for reference..
+            subscriptions = ServiceSubscription.objects.filter(
+                reference_number=transaction.reference_number
+            )
+
+            if len(subscriptions) > 1:
+                logger.warn(
+                    f"Suspicious: more than one service subscription with same reference!"
+                )
```

Comment:
or should this even abort doing anything?

Reply:
Maybe. Should never happen.